---
hide_table_of_contents: true
hide_title: true
custom_edit_url: null
---

import CenterLayout from "/src/components/CenterLayout"
import GitHub from "/src/components/GitHub"

<CenterLayout>
<span className="searchCategory">Knowledge Base</span>
<h1>EKS Kubernetes can&apos;t pull from ECR</h1>
<GitHub discussion={{"id":"D_kwDOF8slf84AO_Fa","number":265,"author":{"login":"yorinasub17"},"title":"EKS Kubernetes can't pull from ECR","body":"I am using Fargate workers and getting the following error launching pods:\r\n\r\n```\r\nFailed to pull image \"111122223333.dkr.ecr.us-west-2.amazonaws.com/my-repo:latest\": rpc error: code = Unknown desc = faile │\r\n│ d to pull and unpack image \"111122223333.dkr.ecr.us-west-2.amazonaws.com/my-repo:latest\": failed to resolve reference \"111122223333.dkr.ecr.us-west-2.amazonaws.com/my-repo:latest\": failed  │\r\n│ to do request: Head \"https://111122223333.dkr.ecr.us-west-2.amazonaws.com/v2/my-repo/manifests/latest\": dial tcp 10.0.166.164:443: i/o timeout \r\n```\r\n\r\nNote that we were able to successfully deploy Pods in the past and this only suddenly started happening recently. Note that the repo has the correct associated cross account permissions set.","bodyHTML":"<p dir=\"auto\">I am using Fargate workers and getting the following error launching pods:</p>\n<div class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"Failed to pull image &quot;111122223333.dkr.ecr.us-west-2.amazonaws.com/my-repo:latest&quot;: rpc error: code = Unknown desc = faile │\n│ d to pull and unpack image &quot;111122223333.dkr.ecr.us-west-2.amazonaws.com/my-repo:latest&quot;: failed to resolve reference &quot;111122223333.dkr.ecr.us-west-2.amazonaws.com/my-repo:latest&quot;: failed  │\n│ to do request: Head &quot;https://111122223333.dkr.ecr.us-west-2.amazonaws.com/v2/my-repo/manifests/latest&quot;: dial tcp 10.0.166.164:443: i/o timeout \"><pre class=\"notranslate\"><code>Failed to pull image \"111122223333.dkr.ecr.us-west-2.amazonaws.com/my-repo:latest\": rpc error: code = Unknown desc = faile │\n│ d to pull and unpack image \"111122223333.dkr.ecr.us-west-2.amazonaws.com/my-repo:latest\": failed to resolve reference \"111122223333.dkr.ecr.us-west-2.amazonaws.com/my-repo:latest\": failed  │\n│ to do request: Head \"https://111122223333.dkr.ecr.us-west-2.amazonaws.com/v2/my-repo/manifests/latest\": dial tcp 10.0.166.164:443: i/o timeout \n</code></pre></div>\n<p dir=\"auto\">Note that we were able to successfully deploy Pods in the past and this only suddenly started happening recently. Note that the repo has the correct associated cross account permissions set.</p>","answer":{"body":"```\r\ndial tcp 10.0.166.164:443: i/o timeout\r\n```\r\n\r\nIndicates that this may be a routing issue. The `10.0.166.164` indicates that it is using [a VPC private link](https://docs.aws.amazon.com/AmazonECR/latest/userguide/vpc-endpoints.html) to access the ECR endpoint. VPC private links can have their own ACLs, which may be preventing access.\r\n\r\nDid you recently add a VPC Private Link endpoint for ECR (our Service Catalog module does not deploy a VPC endpoint for ECR by default), and if so, did you add the subnets that Fargate uses to be able to access it?\r\n\r\nEDIT: This turned out to be the problem and VPC endpoints were added for ECR without the Fargate subnets being added to it. Once the Fargate subnets were added to the ACL, it started to work again.","bodyHTML":"<div class=\"snippet-clipboard-content notranslate position-relative overflow-auto\" data-snippet-clipboard-copy-content=\"dial tcp 10.0.166.164:443: i/o timeout\"><pre class=\"notranslate\"><code>dial tcp 10.0.166.164:443: i/o timeout\n</code></pre></div>\n<p dir=\"auto\">Indicates that this may be a routing issue. The <code class=\"notranslate\">10.0.166.164</code> indicates that it is using <a href=\"https://docs.aws.amazon.com/AmazonECR/latest/userguide/vpc-endpoints.html\" rel=\"nofollow\">a VPC private link</a> to access the ECR endpoint. VPC private links can have their own ACLs, which may be preventing access.</p>\n<p dir=\"auto\">Did you recently add a VPC Private Link endpoint for ECR (our Service Catalog module does not deploy a VPC endpoint for ECR by default), and if so, did you add the subnets that Fargate uses to be able to access it?</p>\n<p dir=\"auto\">EDIT: This turned out to be the problem and VPC endpoints were added for ECR without the Fargate subnets being added to it. Once the Fargate subnets were added to the ACL, it started to work again.</p>"}}} />

</CenterLayout>
  

<!-- ##DOCS-SOURCER-START
{
  "sourcePlugin": "github-discussions",
  "hash": "592a1c2bd3abf9508f6ca7ff4ad3c511"
}
##DOCS-SOURCER-END -->
